src/data/fin.lean

```diff
@@ -63,9 +63,17 @@ open fin nat function
 /-- Elimination principle for the empty set `fin 0`, dependent version. -/
 def fin_zero_elim {α : fin 0 → Sort u} (x : fin 0) : α x := x.elim0
 
+instance fact.succ.pos {n} : fact (0 < succ n) := zero_lt_succ _
+instance fact.bit0.pos {n} [h : fact (0 < n)] : fact (0 < bit0 n) :=
+nat.zero_lt_bit0 $ ne_of_gt h
```

Comment:
I thought we decided no global fact instances that depend on other fact instances?

We don't want to accidentally turn typeclass inference into a prover within a prover. (Yes, I know I am the pot calling the kettle black. :-)

Reply:
This one is a time traveller from before `fact` was a fact of mathlib (I stand by the pun). I'll change it to fit more modern guidelines.